### DISCUSSION TITLE
Is training dataset supposed to be made public as per the Ultralytics AGPL license

### DISCUSSION THREAD

**Discussion by nehagianchandani** [ultralytics/ultralytics#15431] [General]
I read that "all Ultralytics YOLO-trained models fall under the AGPL-3.0 License. The AGPL-3.0 License covers the training code and the models produced by that training code." Does that also mean that when YOLOV8 model trained using the provided by ultralytics but on a private dataset, the training dataset will have to be made public?

I understand that the license applies to the source code and any applications that use the models trained etc, but does it also include the actual data used to train and validate the model?

**Comment by pderrenger** [comment]
@nehagianchandani the AGPL-3.0 license applies to the source code and the models produced by the training code, but it does not require you to make your private training dataset public. The license mandates that if you use Ultralytics models or code, you must open-source your entire project under the same AGPL-3.0 license if you distribute it. However, your private dataset remains your proprietary data and does not need to be shared publicly. For more details, you can refer to the license documentation or contact the Ultralytics licensing team.

**Comment by glenn-jocher** [comment]
@nehagianchandani the AGPL-3.0 license applies to the source code and the models produced by the training code, but it does not require you to make your private training dataset public. The license mandates that if you use the Ultralytics models or code, any modifications or derivative works must also be open-sourced under the same AGPL-3.0 license. However, your training data remains your proprietary asset and does not need to be disclosed. For more details, you can refer to the [Ultralytics license page](https://ultralytics.com/license).
